File: app/schemas/override.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime
import uuid


class OverrideBase(BaseModel):
    resource_id: uuid.UUID
    desired_state: str = Field(..., pattern="^(RUNNING|STOPPED)$")
    expires_at: datetime
    reason: Optional[str] = None


class OverrideCreate(OverrideBase):
    pass


class Override(OverrideBase):
    id: uuid.UUID
    organization_id: uuid.UUID
    created_by_user_id: uuid.UUID
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/lib/types/override.ts
export interface Override {
  id: string;
  organization_id: string;
  resource_id: string;
  desired_state: 'RUNNING' | 'STOPPED';
  expires_at: string;
  reason: string | null;
  created_by_user_id: string;
  created_at: string;
}

export interface OverrideCreate {
  resource_id: string;
  desired_state: 'RUNNING' | 'STOPPED';
  expires_at: string;
  reason?: string;
}
